conan/api/subapi/list.py

```diff
@@ -82,15 +82,15 @@ def packages_configurations(self, ref: RecipeReference,
             packages = app.remote_manager.search_packages(remote, ref)
         return packages
 
-    def filter_packages_configurations(self, pkg_configurations, query):
+    @staticmethod
+    def filter_packages_configurations(pkg_configurations, query):
         """
         :param pkg_configurations: Dict[PkgReference, PkgConfiguration]
         :param query: str like "os=Windows AND (arch=x86 OR compiler=gcc)"
         :return: Dict[PkgReference, PkgConfiguration]
         """
         return filter_packages(query, pkg_configurations)
 
-    # TODO: could it be possible to merge this with subapi.search.select?
     def select(self, pattern, package_query=None, remote=None):
```

Comment:
it's not currently `@staticmethod`, so it's a bit unclear, does it actually modify the state of `self` object? from the name, I suppose, it shouldn't. it throws some exceptions in the middle of the work (e.g. `NotFoundException`), is it possible that function leaves an object in partially modified state?